AR-Top-MobileApp/Assets/Prefabs/JSONReader.cs

```diff
@@ -1,2450 +1,153 @@
+				colorize (tile, obj.color);
 			}
 		}
-		/*for (int k = 0; k <= height; k++) {
-			if (k % gridHeightGap == 0) {
-				Vector3 gridVector = new Vector3 (getGridSize (row) - 1, k - .52f, getGridSize (col) - 1);
-				GameObject gridOutline = Instantiate (gridPrefab, gridVector, Quaternion.identity);
-				gridOutline.transform.SetParent (gridLayout.transform);
-				gridOutline.transform.localScale = new Vector3 (row, 1, col);
-				SpriteRenderer sprite = gridOutline.GetComponentInChildren <SpriteRenderer> ();
-				sprite.color = new Color (0f, 0f, 0f, 1f);
-			}
-		}*/
 
-		foreach (GridItem model in obj.models) {
+		foreach (GridModel model in obj.models)
 			buildPiece (model);
-		}
 
-		Vector3 position = new Vector3 ((mapScaler.transform.position.x - (obj.width / 2))*.1f, mapScaler.transform.position.y*.1f, (mapScaler.transform.position.z - (obj.depth / 2))*.1f);
-		mapScaler.transform.Translate (position);
-		StartCoroutine (scaleMap ());
-		mapScaler.transform.SetParent (imageTarget.transform);
+		Vector3 position = new Vector3 ((mapLayer.transform.position.x - (obj.width / 2))*.1f, mapLayer.transform.position.y*.1f, (mapLayer.transform.position.z - (obj.depth / 2))*.1f);
+		mapLayer.transform.Translate (position);
+		mapLayer.transform.localScale = this.gameObject.transform.localScale;
+		mapLayer.transform.SetParent (this.gameObject.transform);
 
 	}
 
-	IEnumerator scaleMap(){
-		yield return new WaitForSeconds (0.1f);
-		mapScaler.transform.localScale = new Vector3 (0.1f, 0.1f, 0.1f);
-	}
-
 	//Builds a piece based on its type. Places the piece and gives it its color.
-	void buildPiece (GridItem obj)
+	void buildPiece (GridModel obj)
 	{
-		
-		GridItem piece = JsonUtility.FromJson<GridItem> (JsonUtility.ToJson (obj)); //Makes the object a JSON serialized object.
 
 		//Checks the type of the piece. Will be converted to a switch statement to check for all types.
-		switch (piece.type) {
+		switch (obj.type) 
+		{
 
 		case "voxel": 
-			Vector3 tileVector = piece.position; //Tile's position.
-			GameObject tilePiece = Instantiate (tilePrefab, tileVector, Quaternion.identity); //Creates a tile prefab.
-			tilePiece.transform.SetParent (tileLayout.transform); //Sets parent to TileLayout GameObject.
-			fillColor (piece, tilePiece); 
-			break;
-
-		case "character":
-			Vector3 charVector = piece.position; //Character's position.
-			GameObject character = Instantiate (charPrefab, charVector, Quaternion.identity); //Creates a character prefab.
-			character.transform.SetParent (playerLayout.transform); //Sets parent to PlayerLayout GameObject.
-			fillColor (piece, character);
-			break;
-		
-
-		case "npc": 
-			Vector3 npcVector = piece.position; //Character's position.
-			GameObject npc = Instantiate (npcPrefab, npcVector, Quaternion.identity); //Creates a character prefab.
-			npc.transform.SetParent (npcLayout.transform); //Sets parent to PlayerLayout GameObject.
-			fillColor (piece, npc);
```

Comment:
Can we leave this part in?

Reply:
The only thing is that this code is not in use, and very likely to change. In the end, there likely won't be a single model for players.

Reply:
It will be very easy to add back in if we ever need it. we'll just look at the history of this file.